roles/jd-server/src/lib/core.rs

```diff
@@ -0,0 +1,183 @@
+            task::spawn(async move {
+                loop {
+                    let update_mempool_result: Result<(), mempool::error::JdsMempoolError> =
+                        mempool::JDsMempool::update_mempool(mempool_cloned_.clone()).await;
+                    if let Err(err) = update_mempool_result {
+                        match err {
+                            JdsMempoolError::EmptyMempool => {
+                                if last_empty_mempool_warning.elapsed().as_secs() >= 60 {
+                                    warn!("{:?}", err);
+                                    warn!("Template Provider is running, but its mempool is empty (possible reasons: you're testing in testnet, signet, or regtest)");
+                                    last_empty_mempool_warning = std::time::Instant::now();
+                                }
+                            }
+                            JdsMempoolError::NoClient => {
+                                mempool::error::handle_error(&err);
+                                handle_result!(sender_update_mempool, Err(err));
+                            }
+                            JdsMempoolError::Rpc(_) => {
+                                mempool::error::handle_error(&err);
+                                handle_result!(sender_update_mempool, Err(err));
+                            }
+                            JdsMempoolError::PoisonLock(_) => {
+                                mempool::error::handle_error(&err);
+                                handle_result!(sender_update_mempool, Err(err));
+                            }
+                        }
+                    }
+                    tokio::time::sleep(mempool_update_interval).await;
+                    // DO NOT REMOVE THIS LINE
+                    //let _transactions = mempool::JDsMempool::_get_transaction_list(mempool_cloned_.clone());
+                }
+            });
+
+            let mempool_cloned = mempool.clone();
+            let sender_submit_solution = sender.clone();
+            task::spawn(async move {
+                loop {
+                    let result = mempool::JDsMempool::on_submit(mempool_cloned.clone()).await;
+                    if let Err(err) = result {
+                        match err {
+                            JdsMempoolError::EmptyMempool => {
+                                if last_empty_mempool_warning.elapsed().as_secs() >= 60 {
+                                    warn!("{:?}", err);
+                                    warn!("Template Provider is running, but its mempool is empty (possible reasons: you're testing in testnet, signet, or regtest)");
+                                    last_empty_mempool_warning = std::time::Instant::now();
+                                }
+                            }
+                            _ => {
+                                // TODO here there should be a better error managmenet
```

Comment:
Strong indications that there is a need for a JDS specific error handling review. 😅